### ISSUE TITLE
Entry validator not updating if content is changed via data binding after SetContent

### ISSUE THREAD

**Issue by Cookie04DE** [fyne-io/fyne#2639]
### Describe the bug:

If you create a form and put in an entry with data, set a custom validator and change the data after you set the content of the window the entry will display an error even though the content is correct according to the validator and you won't be to submit the form.

### To Reproduce:
Steps to reproduce the behaviour:
1. Copy the example code below
2. Run it and see the red outlining and the inability to submit the form
3. Swap the lines `integerBind.Set(1)`with `w.SetContent(f)`
4. Run it again and see that it works now

### Screenshots:

With the SetContent line before the Set line:
![before](https://user-images.githubusercontent.com/52695768/141339441-7a7a62bc-ac21-424b-a21c-5993b0608650.png)
With the SetContent line after the Set line:
![after](https://user-images.githubusercontent.com/52695768/141339627-aafd17fe-9d32-4e61-ad34-41c6c2c2ddc1.png)

### Example code:

```go
package main

import (
	"fmt"
	"strconv"

	"fyne.io/fyne/v2/app"
	"fyne.io/fyne/v2/data/binding"
	"fyne.io/fyne/v2/widget"
)

func main() {
	a := app.New()
	w := a.NewWindow("Test")

	var integer int

	integerBind := binding.BindInt(&integer)

	dayEntry := widget.NewEntryWithData(binding.IntToString(integerBind))
	dayEntry.Validator = func(s string) error {
		i, err := strconv.Atoi(s)
		if err != nil {
			return fmt.Errorf("Must be an integer")
		}
		if i < 1 {
			return fmt.Errorf("Must be bigger than zero")
		}
		return nil
	}
	f := widget.NewForm(
		widget.NewFormItem("Integer", dayEntry),
	)
	f.SubmitText = "Submit"
	f.OnSubmit = func() {
	}
	w.SetContent(f)
	integerBind.Set(1) //works if this line is placed before the call to SetContent
	w.ShowAndRun()
}
```

### Device (please complete the following information):
 - **OS:** MacOS
 - **Version:** Monterey 12.0.1
 - **Go version:** go1.17.1 darwin/arm64
 - **Fyne version:** 2.1.1

**Comment by drognisep** [comment]
I think we're running into the same thing, @Cookie04DE.

The screenshots show that the value in the `Entry` *is* updated, but the validator seems to not like it because the submit button is disabled. Changing the string in the form item in any way re-runs the validator and the submit button is enabled.

@andydotxyz or @Jacalz have you all had a chance to look at this? This code exhibits the same behavior.
```go
package main

import (
	"fyne.io/fyne/v2"
	"fyne.io/fyne/v2/app"
	"fyne.io/fyne/v2/data/binding"
	"fyne.io/fyne/v2/data/validation"
	"fyne.io/fyne/v2/dialog"
	"fyne.io/fyne/v2/widget"
)

func main() {
	w := app.New().NewWindow("Testing form validation")
	w.CenterOnScreen()
	w.Resize(fyne.NewSize(300, 100))

	someString := ""
	boundString := binding.BindString(&someString)
	entry := widget.NewEntryWithData(boundString)
	entry.Validator = validation.NewRegexp(`^[a-z]+$`, "Must be lowercase characters")
	item := widget.NewFormItem("Some String", entry)
	form := widget.NewForm(item)

	form.OnSubmit = func() {
		dialog.ShowInformation("It worked!", "It worked!", w)
	}

	w.SetContent(form)
	_ = boundString.Set("test")
	w.ShowAndRun()
}

```

![image](https://user-images.githubusercontent.com/8890988/141698543-a104e135-a481-4534-a32f-13db6f4382c6.png)
![image](https://user-images.githubusercontent.com/8890988/141698553-6f586c16-450e-4c06-8bc4-1ae41cf30f11.png)
![image](https://user-images.githubusercontent.com/8890988/141698597-0ad6dd18-c843-4d9a-bc9f-64e62f236b02.png)

Using a form dialog instead shows the same behavior.

![image](https://user-images.githubusercontent.com/8890988/141698660-5cd393e6-7b65-478b-8eca-5c3d7f9f9c6a.png)

**Comment by drognisep** [comment]
Interesting! It seems that if the value is already valid, then the entry is still shown as valid even if it's set to a value that would not be accepted by the validator. In my code above I initialized `someString` to `"a"` and used `_ = boundString.Set("123")` instead.

![image](https://user-images.githubusercontent.com/8890988/141698887-19b5dda3-37f4-4c9e-b2c2-c544f1d5b5e2.png)

So the root of this seems to be that the validator is not necessarily called when a binding is `Set`.

**Comment by andydotxyz** [comment]
I am pretty sure this has been resolved on `release/v2.1.x` through work on another bug. I could replicate it before, but cannot now.
